sdk/core/Azure.Core/tests/TestFramework/RecordSession.cs

```diff
@@ -78,14 +78,14 @@ public void Sanitize(RecordedTestSanitizer sanitizer)
         {
             lock (Entries)
```

Comment:
Let's move this method into `RecordedTestSanitizer` instead so you can override it completely and make the logic less fragile.